### PR TITLE
Tidy up several parts of the batching code.

### PR DESCRIPTION
**Pull request by glennw**
This is prep work for some of the upcoming border refactoring.

None of the diffs in this file should result in any behaviour
changes.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1073)

### CODE REVIEW

**Comment by glennw** [comment]
r? @kvark 

Sorry the patch looks quite large - the majority of it is just shuffling code around, so it should hopefully be easy to review.

**Comment by glennw** [comment]
I've run against the WPT/CSS tests successfully.

File: webrender/src/tiling.rs [hunk 1]
```diff
@@ -714,28 +730,17 @@ impl AlphaBatcher {
...
-                        let needs_clipping_flag = if needs_clipping {
-                            NEEDS_CLIPPING
-                        } else {
-                            AlphaBatchKeyFlags::empty()
-                        };
-                        let flags = match transform_kind {
-                            TransformedRectKind::AxisAligned => AXIS_ALIGNED | needs_clipping_flag,
-                            _ => needs_clipping_flag,
-                        };
+                        debug_assert!(!prim_metadata.needs_clipping());
```
Comment (kvark) [inline webrender/src/tiling.rs]: I'd rather not tie up clipping, axis-alignment, and opaque-ness properties together just yet.

**Comment by kvark** [comment]
The code is much prettier now!
@bors-servo r+

**Pull request merged**